### PR TITLE
[#5451] [desktop] fix flickering topic input

### LINKED ISSUE #5451: Text in public chat name starts flickering when quickly typed
### Description

*Type*: Bug
*Summary*: when you quickly type name in "Join Public Chat"  name starts flickering without any user interaction; CPU usage jumps to ~70-90%.

#### Expected behavior
name is shown

#### Actual behavior
name starts flickering without any user interaction; CPU usage jumps to ~70-90%.

### Reproduction

- Open Status
- Click on +
- Set focus to field below "Join public chat" and quickly (as much as you can) type any text (`a`-`z` chars)

### Additional Information
[comment]: # (Please do your best to fill this out.)
* Status version: nightly 10/08/2018
* Operating System: Mac OSx High Sierra
* Video: http://take.ms/fqqOC

#### Logs

```
Debug: ubuntu-server std:  "DEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic"

Debug: ubuntu-server std:  "DEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic"

Debug: ubuntu-server std:  "DEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic"

Debug: ubuntu-server std:  "DEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic"

Debug: ubuntu-server std:  "DEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic"

Debug: ubuntu-server std:  "DEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic"

Debug: ubuntu-server std:  "DEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic"

Debug: ubuntu-server std:  "DEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic"

Debug: ubuntu-server std:  "DEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic"

Debug: ubuntu-server std:  "DEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic\nDEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic"

Debug: ubuntu-server std:  "DEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic"

Debug: ubuntu-server std:  "DEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic"

Debug: ubuntu-server std:  "DEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic"

Debug: ubuntu-server std:  "DEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic"

Debug: ubuntu-server std:  "DEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic"

Debug: ubuntu-server std:  "DEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic"

Debug: ubuntu-server std:  "DEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic"

Debug: ubuntu-server std:  "DEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic"

Debug: ubuntu-server std:  "DEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic"

Debug: ubuntu-server std:  "DEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic"

Debug: ubuntu-server std:  "DEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :set :public-group-topic"

```

### PR DESCRIPTION
**Pull request by rasom**
fix #5451

status: ready

### CODE REVIEW

File: src/status_im/ui/screens/desktop/main/add_new/views.cljs [hunk 1]
```diff
@@ -68,20 +71,15 @@
...
-                                                     [_ before after] (first (re-seq #"(.*)\#(.*)" text))]
-                                                 (.setNativeProps @topic-input-ref (js-obj "text" (str "#" before after)))
-                                                 (re-frame/dispatch [:set :public-group-topic (subs text 1)])))}]]
+       [react/view {:flex  1
+                    :style styles/add-pub-chat-input}
+        [react/text-input {:flex            1
+                           :font            :default
+                           :selection-color colors/hawkes-blue
+                           :placeholder     ""
+                           :on-change       (fn [e]
```
Comment (yenda) [inline src/status_im/ui/screens/desktop/main/add_new/views.cljs]: could be put in a separate function

**Review by janherich — APPROVED** [review]

**Review by yenda — APPROVED** [review]

File: src/status_im/ui/screens/desktop/main/add_new/views.cljs (hunk 1 above)
Comment (rasom) [reply src/status_im/ui/screens/desktop/main/add_new/views.cljs]: done

**Review by vkjr — APPROVED** [review]

**Comment by vkjr** [comment]
@rasom, what was the reason of flickering? Would be glad to know to avoid this mistake in future.

**Comment by yenda** [comment]
@vkjr trying to update the content with the value from app-db. There is no need for that

edit: the part about app-db is wrong, it's just trying to update the content of the input field

**Review by vitvly — APPROVED** [review]

**Comment by rasom** [comment]
@yenda no, the reason is different
@vkjr there was a race condition in onChange handler

```clojure
(fn [topic-input-ref]
  (let [native-event (.-nativeEvent e)
        text         (.-text native-event)
        [_ before after] (first (re-seq #"(.*)\#(.*)" text))]
    (.setNativeProps @topic-input-ref (js-obj "text" (str "#" before after)))
    (re-frame/dispatch [:set :public-group-topic (subs text 1)])))
```

so `(first (re-seq #"(.*)\#(.*)" text))` is quite slow, and when i type "1" then "2" fastly happens the next thing
1) `text` prop changed to `"1"` by pressing keyboard
2) re-seq in progress
3) `text` prop changed to `"12"` by pressing keyboard
4) re-seq finished and `text` prop is changed to `"1"` back by ` (.setNativeProps @topic-input-ref (js-obj "text" (str "#" before after)))`
5) and then `onChange` is called again because `text` has been changed from `"12"` to `"1"`
6) meanwhile `re-seq` finished for `"12"`
7) and here we go in circle :)

**Pull request merged**